### PR TITLE
fix: alertmanager: validate labels and annotation names only, not values

### PR DESCRIPTION
**Pull request by hileef**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area config

**What this PR does / why we need it**:

Only validate Prometheus Label **names** and Annotation **names** _instead of validating **values** as well_.
Internally, Alertmanager and Prometheus use the same data model to represent both labels and annotations,
and apply the same validation logic.

See official doc here : https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels, which mentions : 

> Label names may contain ASCII letters, numbers, as well as underscores. They must match the regex [a-zA-Z_][a-zA-Z0-9_]*. Label names beginning with __ are reserved for internal use.
> 
> Label values may contain any Unicode characters.

As a practical example, we encountered an issue because we use dashes in our label values,
such that this was rejected : `extralabels: "cloud:aws,region:eu-west-1,team:platform"`

**Special notes for your reviewer**:

Hello, we are currently in the process of experimenting with this project for internal usage ; 
it looks quite promising, great work 💪 😄 

We encountered an issue when trying to use our standardized internal alert routing,
which should be fixed with these changes 🙏 

It wasn't obvious to me where to add a test for such a bug fix,
I may have missed something but could not find config validation tests
so I didn't add one for this specific use case, I hope that's okay.

### CODE REVIEW

**Review by Issif — APPROVED** [review]
Thanks and sorry for the delay I was sick for past few days

**Comment by hileef** [comment]
> Thanks and sorry for the delay I was sick for past few days

No problem 🙏 Glad to hear you are doing better @Issif  🙂 

I'm not sure how to proceed with this PR though, I do notice that there are "3 workflows awaiting approval" 🤔 
Is this PR experiencing something similar to https://github.com/falcosecurity/falcosidekick/pull/431#issuecomment-1483843121 ? 

Is there something I can do to help ? 🙂

**Comment by Issif** [comment]
Thanks.

The merge should be automatic after my approval, let me check

**Pull request merged**
